src/industry_gui.cpp

```diff
@@ -2348,15 +2348,15 @@ struct IndustryCargoesWindow : public Window {
 		this->ind_cargo = it;
 		_displayed_industries.reset();
 		_displayed_industries.set(it);
-
-		this->fields.Clear();
-		CargoesRow *row = this->fields.Append();
-		row->columns[0].MakeHeader(STR_INDUSTRY_CARGOES_PRODUCERS);
-		row->columns[1].MakeEmpty(CFT_SMALL_EMPTY);
-		row->columns[2].MakeEmpty(CFT_SMALL_EMPTY);
-		row->columns[3].MakeEmpty(CFT_SMALL_EMPTY);
-		row->columns[4].MakeHeader(STR_INDUSTRY_CARGOES_CUSTOMERS);
-
+		{
```

Comment:
why the addition of a scope level here?

Reply:
@LordAro It scopes the variable `row` and prevents it to leak into the rest of the code. So, it prevents headaches.
Ultimately, this scope should really be a separate function though.